src/main/java/org/mifos/processor/bulk/camel/processor/AccountLookupCallbackProcessor.java

```diff
@@ -0,0 +1,60 @@
+package org.mifos.processor.bulk.camel.processor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.zeebe.client.ZeebeClient;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+import org.mifos.processor.bulk.schema.AccountLookupResponseDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mifos.processor.bulk.camel.config.CamelProperties.*;
+import static org.mifos.processor.bulk.zeebe.ZeebeMessages.ACCOUNT_LOOKUP;
+import static org.mifos.processor.bulk.zeebe.ZeebeVariables.*;
+
+@Component
+public class AccountLookupCallbackProcessor implements Processor {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired(required = false)
+    private ZeebeClient zeebeClient;
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        Map<String, Object> variables = new HashMap<>();
+
+        String error = null;
+        String response = exchange.getIn().getBody(String.class);
+        AccountLookupResponseDTO accountLookupResponseDTO = null;
+        try {
+            accountLookupResponseDTO = objectMapper.readValue(response, AccountLookupResponseDTO.class);
+        }catch(IOException e){
+            variables.put(ACCOUNT_LOOKUP_FAILED, true);
+        }
+
+        variables.put(ACCOUNT_LOOKUP_FAILED, false);
+        variables.put(PAYEE_PARTY_ID, response != null ? accountLookupResponseDTO.getPaymentModalityList().get(0).getFinancialAddress() : null);
```

Comment:
what if resonse is not nill but the Json parsing failed? this line will throw null pointer exception accountLookupResponseDTO.getPaymentModalityList(). make sure you compare the right  variable for null check.

Same issue in below three lines as well